t/sys/chdir.c

```diff
@@ -0,0 +1,361 @@
+/*
+ * Copyright (c) 2019, Lawrence Livermore National Security, LLC.
+ * Produced at the Lawrence Livermore National Laboratory.
+ *
+ * Copyright 2018, UT-Battelle, LLC.
+ *
+ * LLNL-CODE-741539
+ * All rights reserved.
+ *
+ * This is the license for UnifyFS.
+ * For details, see https://github.com/LLNL/UnifyFS.
+ * Please read https://github.com/LLNL/UnifyFS/LICENSE for full license text.
+ */
+
+ /*
+  * Test chdir/fchdir/getcwd/getwd/get_current_dir_name
+  */
+#include <fcntl.h>
+#include <string.h>
+#include <errno.h>
+#include <linux/limits.h>
+#include <stdio.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+#include <unistd.h>
+#include <dirent.h>
+#include "t/lib/tap.h"
+#include "t/lib/testutil.h"
+
+int chdir_test(char* unifyfs_root)
+{
+    diag("Starting UNIFYFS_WRAP(chdir/fchdir/getcwd/getwd/"
+         "get_current_dir_name) tests");
+
+    char path[64];
+    int rc;
+    char* str;
+
+    errno = 0;
+
+    testutil_rand_path(path, sizeof(path), unifyfs_root);
+
+    /* define a dir1 subdirectory within unifyfs space */
+    char buf[64] = {0};
+    snprintf(buf, sizeof(buf), "%s/dir1", unifyfs_root);
+    rc = mkdir(buf, 0700);
+    ok(rc == 0 || errno == EEXIST, "%s:%d mkdir(%s): %s",
+       __FILE__, __LINE__, buf, strerror(errno));
+    errno = 0;
+
+    /* define a dir2 subdirectory within unifyfs space */
+    char buf2[64] = {0};
+    snprintf(buf2, sizeof(buf2), "%s/dir2", unifyfs_root);
+    rc = mkdir(buf2, 0700);
+    ok(rc == 0 || errno == EEXIST, "%s:%d mkdir(%s): %s",
+       __FILE__, __LINE__, buf2, strerror(errno));
+    errno = 0;
+
+    /* change to root directory */
+    rc = chdir("/");
+    ok(rc == 0, "%s:%d chdir(%s): %s",
+       __FILE__, __LINE__, "/", strerror(errno));
+
+    /* check that we're in root directory */
+    str = getcwd(path, sizeof(path));
+    ok(str != NULL, "%s:%d getcwd: %s",
+       __FILE__, __LINE__, strerror(errno));
+    ok(str != NULL && strcmp(str, "/") == 0,
+       "%s:%d getcwd returned %s expected %s",
+       __FILE__, __LINE__, str, "/");
```

Comment:
More just an FYI, but liptap does have a string comparison test. Not necessary though.
```suggestion
    str = getcwd(path, sizeof(path));
    ok(str != NULL, "%s:%d getcwd: %s",
       __FILE__, __LINE__, strerror(errno));
    is(str, "/",
       "%s:%d getcwd returned %s expected %s",
       __FILE__, __LINE__, str, "/");
```

Reply:
Nice, I'll switch since that's cleaner.  Can we also pass a variable name in place of a hard-coded string, like ```is(str, unifyfs_root, ...);```?

Reply:
Yeah, should be able to. I haven't tested it myself using two variables though. Tony used it here, but only with one variable name
https://github.com/LLNL/UnifyFS/blob/1d12d24a7710dfd5ba7bc096993fffb1891f3ff2/t/sys/write-read.c#L152

Reply:
Yep, variable names are fine.